native_client/javascript/stt.i

```diff
@@ -47,7 +59,8 @@ using namespace node;
 }
 
 %typemap(argout) ModelState **retval {
-  $result = SWIGV8_ARRAY_NEW(0);
+  //TODO: check if the swig API is updated
+  $result = SWIGV8_ARRAY_NEW();
```

Comment:
I dont remember precisely, and it might be due to how we hacked swig back then, but clearly I have some memory of tracking weird crashes related to the absence of `0` here.

Reply:
Perhaps I could remove these changes and we could run the CI to see if it compiles then? It feels like something setup-related to me.

Reply:
@lissyx i reverted the SWIGV8_ARRAY_NEW changes, as I assume the CI build is working and this is probably related to my setup.

Reply:
Why did you need the change in the first place

Reply:
I was unable to compile the bindings from the main branch without these changes. Using the provided swig version, I got the following error:
`error: macro "SWIGV8_ARRAY_NEW" passed 1 arguments, but takes just 0`.
This hack is what allowed me to build and then develop the loading from buffer part.

Reply:
I'd be very skeptical regarding the swig build you used. Since I'm not working on DeepSpeech anymore, I have lost track of the changes we tried to push upstream for newer versions of NodeJS / V8, but it was my understanding maybe not everything might have been picked yet. When you say "provided version", can you be explicit on where it comes from ?

Reply:
SWIG is handled by https://github.com/coqui-ai/STT/blob/20650769881b40ba1a770622943ae182fa5ad5b6/.github/workflows/build-and-test.yml#L82-L169

Reply:
So https://github.com/swig/swig/commit/90cdbee6a69d13b39d734083b9f91069533b0d7b which is from March 2021. I'd be not surprised that it is indeed outdated and could have some incompatibility. Yet, https://github.com/coqui-ai/STT/blob/20650769881b40ba1a770622943ae182fa5ad5b6/.github/workflows/build-and-test.yml#L447 shows up to NodeJS v17.

Reply:
You should verify if your `swig` binary is named `ds-swig` and it is indeed built from `90cdbee6a69d13b39d734083b9f91069533b0d7b`

Reply:
I was able to reproduce the error on commit `20650769881b40ba1a770622943ae182fa5ad5b6` (the latest one on the main branch). 
I can provide the output from the build command here if you wish. I could confirm that the swig -swiglib step is outputting `<STT_REPO_ROOT_DIR>/native_client/ds-swig/share/swig/4.1.0/`, but I could not find a way to confirm that it was built from `90cdbee6a69d13b39d734083b9f91069533b0d7b`.
I used node v14.19.1, should I try using the 14.0.0 release perhaps?

Reply:
It's unlikely 14.0.0 vs 14.9.1 makes a difference from my memory of V8 headers, but it's still a good idea to check. Maybe try to remove the `ds-swig` and let tooling re-download one ?

@reuben Are you aware of such a regression ?

Reply:
I tried with v14.0.0 but the output was the same. After deleting the ds-swig folder, the tooling downloaded swig from `https://github.com/mozilla/DeepSpeech/releases/download/v0.9.3/ds-swig.linux.amd64.tar.gz`, is that the right version? I could not find swig on the current coqui release.

Reply:
Those URLs are still here, https://github.com/coqui-ai/STT/blob/20650769881b40ba1a770622943ae182fa5ad5b6/native_client/definitions.mk#L210-L221 so I really defer to @reuben, he knows that :)

Reply:
Oh, I can see that there is a patched swig version on the 1.3.0 release of STT. I will see if manually downloading that one can fix my problem :)

Reply:
It is building now! Could it be that the swig URLs in `https://github.com/coqui-ai/STT/blob/main/native_client/definitions.mk#L213` are outdated then?

Reply:
> It is building now! Could it be that the swig URLs in `https://github.com/coqui-ai/STT/blob/main/native_client/definitions.mk#L213` are outdated then?

That would advocate for it, I suggest you send a PR to update them :)

Reply:
Great, will do!